### PR TITLE
Add nix getting started guide for students

### PR DESCRIPTION
**Pull request by martin-fink**
Should have the basics for students who have never heard of nix to get started.

### CODE REVIEW

File: docs/nix-getting-started.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+### Development Environment
+
+**Flakes or Nix Shells:** Choose either format to define your project dependencies.
+
+#### Flakes (recommended)
+
+If you're inside a git project, `git add flake.nix`, as nix cannot find it otherwise. 
+
+Example `flake.nix`:
```
Comment (Mic92) [inline docs/nix-getting-started.md]: We got these templates that we maybe want to re-use or update: https://github.com/TUM-DSE/doctor-cluster-config/tree/master/templates
this has both a project and a home-manager template.

File: docs/nix-getting-started.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
+# NixOS Getting Started
+
+For information about NFS, where to place your projects. see [this README](./README.md#storage).
+TL;DR: Your home directory is a NFS share, which is backed up. Do not place VM images/projects/... there.
+Use `/scratch/$USER` if you need fast, local disk access. This directory is not backed up.
+
+## Packages
+
+- **Where to look for packages:** https://search.nixos.org
```
Comment (Mic92) [inline docs/nix-getting-started.md:9]:
```suggestion
- **Where to look for packages:** https://search.nixos.org
- Find packages by filename: `nix-locate someheader.h` (nix-locate is installed on our servers)
```

File: docs/nix-getting-started.md (hunk 1 above)
Comment (martin-fink) [reply docs/nix-getting-started.md]: Good point, I'll link to those instead.

File: docs/nix-getting-started.md (hunk 2 above)
Comment (martin-fink) [reply docs/nix-getting-started.md:9]: Thanks!

**Comment by Mic92** [comment]
@mergify rebase

**Comment by Mic92** [comment]
@mergify queue

File: docs/nix-getting-started.md [hunk 3]
```diff
@@ -93,50 +54,7 @@ Add `.direnv` to your `.gitignore`.
...
-{
-  # Let Home Manager install and manage itself.
-  programs.home-manager.enable = true;
-
-  # Home Manager needs a bit of information about you and the
-  # paths it should manage.
-  home.username = "<your name>";
-  home.homeDirectory = "/home/<your name>";
-  xdg.cacheHome = "/scratch/<your name>/.cache";
-  xdg.stateHome = "/scratch/<your name>/.local/share";
```
Comment (Mic92) [inline docs/nix-getting-started.md]: I don't think we have those in our template. But would be good to have, I guess.

Comment (martin-fink) [reply docs/nix-getting-started.md]: I'll create a follow-up PR adding those.

**Pull request merged**